File: leetcode_771_jewels_and_stones.cpp
class Solution {
public:
    int numJewelsInStones(string j, string s) {
        std::unordered_map<char,int>mp;
        for(auto i:s) ++mp[i];
        int cnt=0;
        for(auto c:j){
            cnt+=mp[c];
        }
        return cnt;
    }
};

////////////////////////
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        unordered_set<char> jewelSet(jewels.begin(), jewels.end());
        int count = 0;
        for (char stone : stones) {
            if (jewelSet.find(stone) != jewelSet.end()) {
                count++;
            }
        }
        
        return count;
    }
};

File: java/leetcode_771_jewels_and_stones.java
class Solution {
    public int numJewelsInStones(String j, String s) {
        Map<Character,Integer> mp=new HashMap<>();
        for(Character c:s.toCharArray()){
            mp.put(c,mp.containsKey(c)?mp.get(c)+1:1);
        }
        int cnt=0;
        for(Character c:j.toCharArray()){
            cnt+=mp.containsKey(c)?mp.get(c):0;
        }
        return cnt;
    }
}

/////////////////
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        
        int count=0;
        for(int i=0;i<jewels.length();i++){
            for(int j=0;j<stones.length();j++){
                if(jewels.charAt(i)==stones.charAt(j)){
                    count++;
                }
            }
        }
        return count;
    }
}